plugins/web-worker-offloading/hooks.php

```diff
@@ -68,31 +68,30 @@ function wwo_init(): void {
 add_action( 'wp_enqueue_scripts', 'wwo_init' );
 
 /**
- * Marks scripts with `web-worker-offloading` dependency as async.
+ * Adds `web-worker-offloading` as dependency to scripts with `worker` data. Also, marks their strategy as `async`.
  *
  * This is needed because scripts offloaded to a worker thread can be considered async. However, they may include `before` and `after` inline
  * scripts that need sequential execution. Once marked as async, `filter_eligible_strategies()` determines if the
  * script is eligible for async execution. If so, it will be offloaded to the worker thread.
  *
- * @since 0.1.0
- *
- * @param string[]|mixed $script_handles Array of script handles.
- * @return string[] Array of script handles.
+ * @since n.e.x.t
  */
-function wwo_update_script_strategy( $script_handles ): array {
-	$script_handles = array_intersect( (array) $script_handles, array_keys( wp_scripts()->registered ) );
-	foreach ( $script_handles as $handle ) {
-		if ( in_array( 'web-worker-offloading', wp_scripts()->registered[ $handle ]->deps, true ) ) {
-			if ( false === wp_scripts()->get_data( $handle, 'strategy' ) ) {
-				wp_script_add_data( $handle, 'strategy', 'async' ); // The 'defer' strategy would work as well.
-				wp_script_add_data( $handle, 'wwo_strategy_added', true );
+function wwo_update_worker_scripts_deps_and_strategy(): void {
+	foreach ( wp_scripts()->registered as $dep ) {
+		if (
+			(bool) wp_scripts()->get_data( $dep->handle, 'worker' ) &&
+			! in_array( 'web-worker-offloading', wp_scripts()->registered[ $dep->handle ]->deps, true )
+		) {
+			wp_scripts()->registered[ $dep->handle ]->deps[] = 'web-worker-offloading';
+
+			if ( false === wp_scripts()->get_data( $dep->handle, 'strategy' ) ) {
+				wp_script_add_data( $dep->handle, 'strategy', 'async' ); // The 'defer' strategy would work as well.
+				wp_script_add_data( $dep->handle, 'wwo_strategy_added', true );
 			}
```

Comment:
We may need to end up removing this logic if we are going to rely on inline after scripts also being made Partytown scripts as mentioned in https://github.com/WordPress/performance/issues/1455. But that can be done in another PR.

Reply:
I'll address this in another PR.